src/System.ServiceProcess.ServiceController/tests/System.ServiceProcess.ServiceController.TestService/Program.cs

```diff
@@ -15,8 +15,28 @@ static int Main(string[] args)
         {
             if (args.Length == 1 || args.Length == 2)
             {
-                TestService testService = new TestService(args[0]);
-                ServiceBase.Run(testService);
+                TestService testService;
+                if (args[0].StartsWith("PropagateExceptionFromOnStart"))
+                {
+                    Exception expectedException = new InvalidOperationException("Fail on startup.");
+                    testService = new TestService(args[0], expectedException);
+                    try
+                    {
+                        ServiceBase.Run(testService);
+                    }
+                    catch(Exception actualException)
+                    {
+                        if (object.ReferenceEquals(expectedException, actualException))
+                        {
+                            testService.WriteStreamAsync(PipeMessageByteCode.ExceptionThrown).Wait();
```

Comment:
if it's not expectedExceptoin maybe it should rethrow it?

Reply:
Rethrowing won’t affect the test as it is in other process.